Fix code warnings

We have some warnings in the code:
```
[ 82%] Building C object src/CMakeFiles/g2_d_c.dir/dec_jpeg2000.c.o
cd /home/ed/NCEPLIBS-g2/b/src && /usr/bin/cc -DLINUX -D__64BIT__ -I/usr/local/jasper-2.0.33/include -I/usr/include/libpng16 -g  -ggdb -Wall -MD -MT src/CMakeFiles/g2_d_c.dir/dec_jpeg2000.c.o -MF CMakeFiles/g2_d_c.dir/dec_jpeg2000.c.o.d -o CMakeFiles/g2_d_c.dir/dec_jpeg2000.c.o -c /home/ed/NCEPLIBS-g2/src/dec_jpeg2000.c
/home/ed/NCEPLIBS-g2/src/dec_jpeg2000.c: In function ‘dec_jpeg2000_’:
/home/ed/NCEPLIBS-g2/src/dec_jpeg2000.c:45:22: warning: unused variable ‘cmpt’ [-Wunused-variable]
   45 |     jas_image_cmpt_t cmpt, *pcmpt;
      |                      ^~~~
/home/ed/NCEPLIBS-g2/src/dec_jpeg2000.c:44:31: warning: unused variable ‘istream’ [-Wunused-variable]
   44 |     jas_stream_t *jpcstream, *istream;
      |                               ^~~~~~~
/home/ed/NCEPLIBS-g2/src/dec_jpeg2000.c:42:20: warning: unused variable ‘n’ [-Wunused-variable]
   42 |     g2int i, j, k, n;
      |                    ^

```

and

```
/home/ed/NCEPLIBS-g2/src/enc_png.c:86:9: warning: unused variable ‘do_nothing’ [-Wunused-variable]
   86 |    int *do_nothing=NULL;
      |         ^~~~~~~~~~

```

and

```
/home/runner/work/NCEPLIBS-g2/NCEPLIBS-g2/g2/src/g2grids.f:124:0:

  124 |                 prev%next => gtemp      ! points to the new entry,
      | 

```

--- a/.github/workflows/gcc_docs_test-coverage.yml
+++ b/.github/workflows/gcc_docs_test-coverage.yml
@@ -67,7 +67,7 @@ jobs:
         mkdir build
         doxygen --version
         cd build
-        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall -Werror" -DCMAKE_C_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall -Werror"
+        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall" -DCMAKE_C_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall"
         make -j2 VERBOSE=1 
 
     - name: test

--- a/.github/workflows/gcc_docs_test-coverage.yml
+++ b/.github/workflows/gcc_docs_test-coverage.yml
@@ -67,7 +67,7 @@ jobs:
         mkdir build
         doxygen --version
         cd build
-        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall"
+        cmake .. -DJasper_ROOT=~/Jasper -DCMAKE_PREFIX_PATH=~/bacio -DENABLE_DOCS=Yes -DCMAKE_Fortran_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall -Werror" -DCMAKE_C_FLAGS="-g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall -Werror"
         make -j2 VERBOSE=1 
 
     - name: test

--- a/src/dec_jpeg2000.c
+++ b/src/dec_jpeg2000.c
@@ -39,10 +39,10 @@
 int
 dec_jpeg2000_(char *injpc, g2int *bufsize, g2int *outfld)
 {
-    g2int i, j, k, n;
+    g2int i, j, k;
     jas_image_t *image = NULL;
-    jas_stream_t *jpcstream, *istream;
-    jas_image_cmpt_t cmpt, *pcmpt;
+    jas_stream_t *jpcstream;
+    jas_image_cmpt_t *pcmpt;
     char *opts = NULL;
     jas_matrix_t *data;
     int ier = 0;

--- a/src/enc_png.c
+++ b/src/enc_png.c
@@ -81,9 +81,7 @@ void user_write_data(png_structp png_ptr,png_bytep data, png_uint_32 length)
 */
 
 void user_flush_data(png_structp png_ptr)
-
 {
-   int *do_nothing=NULL;
 }
 
 /**

